crates/mun_runtime/src/assembly.rs

```diff
@@ -356,6 +404,12 @@ impl Assembly {
         &self.info
     }
 
+    /// Returns the assembly's information.
+    pub fn info_mut(&mut self) -> &mut abi::AssemblyInfo {
+        // HACK: We want to make sure that the assembly info never outlives self.
```

Comment:
Why is this necessary now?

Reply:
Lifetimes are a bit mixed up here. When loading the symbol from the library I would like to already give the AssemblyInfo the lifetime of the library it came from, but I couldn’t get that to compile because it would create a self referential lifetime, since the AssemblyInfo is stored next to the library it came from. To work around this issue I give the stored value the static lifetime and then use this function to ensure the rest of the could does use an AssemblyInfo with a properly scoped lifetime.

Reply:
I saw you added a dependency for self-reference. Can that be removed then?

Reply:
Yes! Good one!